### PR TITLE
feat: Add context data to events

### LINKED ISSUE #37: Use better defaults for some options
- **release** - `app.getName()` + `app.getVersion()`
- **environment** - `process.defaultApp ? 'development' : 'packaged'`
- **user.id** - `node-machine-id.machineId()`

### LINKED ISSUE #40: Show correct Electron version in Sentry
We should always send info on Electron framework (name, version, build) and send consistent OS information so that the corresponding context item can be shown in Sentry. Currently, it looks like this:

**Native Crash**:

screen shot 2018-03-14 at 11 31 49

**Renderer Error**:

screen shot 2018-03-14 at 11 31 33

**Main Processes Error**:

*nothing*

### PR DESCRIPTION
**Pull request by jan-auer**
The operating system version is still wrong and I think we can do a better job in getting the operating system name (e.g. via user agent). 

Fixes #37
Fixes #40

### CODE REVIEW

**Comment by jan-auer** [comment]
To properly display those contexts, we still need some changes in the Sentry Context UI. Especially, if we want to provide consistent rendering - see #40 for more details.

File: src/frontend.ts [hunk 1]
```diff
@@ -99,4 +100,15 @@ export class ElectronFrontend extends FrontendBase<
...
+
+  /**
+   * @inheritDoc
+   */
+  protected async prepareEvent(
+    event: SentryEvent,
+    scope: Scope,
+  ): Promise<SentryEvent> {
+    const prepared = await super.prepareEvent(event, scope);
+    return addEventDefaults(prepared);
```
[the remark this answers is not in the document]
Comment (timfish) [reply src/frontend.ts]: Actually its protected and throws from `sendEvent` so this should be fine

[the remark this answers is not in the document]
Comment (jan-auer) [reply src/frontend.ts]: We can add a guard and throw in case someone tries to invoke it 👍

**Comment by jan-auer** [comment]
Updated, this should be ready to merge now.

**Pull request merged**